### PR TITLE
Add arcsine function

### PR DESCRIPTION
**Pull request by abidsikder**
- Updated directory structure to match modern foundry installs
- Added arcsine function in separate .sol file, along with point and fuzz testing

Only issue that remains is that it seems testSinFuzz() seems to be failing.

### CODE REVIEW

File: src/test/trig.py [hunk 1]
```diff
@@ -1,28 +0,0 @@
-"""
```
Comment (mds1) [inline src/test/trig.py]: Could we leave the directory structure as it was for this PR? Prefer keeping refactor PRs like that separate from feature PRs like adding arcsin, makes things easier in the future when looking back at this PR to see what changed

File: src/Arcsin.sol [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  }
+
+  /**
+     * @notice Arcsine function
+     *
+     * @param _x A integer with 18 fixed decimal points, where the whole part is bounded inside of [-1,1]
+     *
+     * @return The arcsine, with 18 fixed decimal points
+     */
+  function arcsin(int256 _x) internal pure returns (int256) {
```
Comment (mds1) [inline src/Arcsin.sol]: What do you think about renaming this file/library to `InverseTrigonometry.sol` then adding an `arccos` function (and tests) based on the fact that `arccos(x) = (pi/2) - arcsin(x)`? (similar to what we do for `cos` in `Trigonometry.sol`. I'm happy to do this after this is merged if you don't have the time, so no pressure

**Review by mds1 — COMMENTED** [review]
This is awesome, thanks so much @abidsikder! LGTM overall. As for the fuzz tests, don't worry about those. They're failing on main too, I need to do a better job at quantifying/testing the errors here, but as long as the concrete test points pass I think this is good for now 👍️

**Comment by abidsikder** [comment]
Both of those changes sound good to me @mds1! You know best how you'd like it to look again, so could you add those reverts/changes?

**Comment by mds1** [comment]
Sounds good, I'll push some commits to this branch this week!

**Comment by mds1** [comment]
Thanks so much @abidsikder! This is great.

Going to merge this now, will fix dir structure / modernize, add arccos, and fix tests in follow up PRs 🙂

**Pull request merged**